app/js/UrlCache.js

```diff
@@ -35,8 +36,12 @@ module.exports = UrlCache = {
         if (error != null) {
           return callback(error)
         }
-        return UrlCache._copyFile(pathToCachedUrl, destPath, function (error) {
+        return fse.copy(pathToCachedUrl, destPath, function (error) {
```

Comment:
Have you thought about calling [`fs.copyFile`](https://nodejs.org/docs/latest-v12.x/api/fs.html#fs_fs_copyfile_src_dest_flags_callback) directly?

fse is doing two extra stat calls we do not need: https://github.com/jprichardson/node-fs-extra/blob/e6a95058c930953113177c9518f57e83cace3e79/lib/copy/copy.js#L30-L36
graceful-fs has some retry queue, which I do not think we want either: https://github.com/isaacs/node-graceful-fs/blob/89dc1330dcd8fa218c5dff92a97d8792b7da6b12/graceful-fs.js#L182-L188

Reply:
I did try that first but found out that fs.copyFile does not create parent directories, unlike fs-extra.copy, so it was not a drop-in replacement.

Reply:
The existing code does not handle this either :shrug: 

```
> fs=require('fs');fs.writeFileSync('/tmp/foo.txt','BLOB');fs.createReadStream('/tmp/foo.txt').on('error',console.error).pipe(fs.createWriteStream('/tmp/does/not/exist.png'))
> Uncaught [Error: ENOENT: no such file or directory, open '/tmp/does/not/exist.png'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/tmp/does/not/exist.png',
```
```

Reply:
Ah, you're right - I've switched it to fs.copyFile.   I got confused about where the error was coming from -- I was experimenting with changing the other use of fse.copy in ResourceWriter to fs.copyFile  at the same time.